daft/expressions2/expressions.py

```diff
@@ -266,8 +266,11 @@ class ExpressionStringNamespace(ExpressionNamespace):
     def contains(self, substr: str) -> Expression:
         raise NotImplementedError("[RUST-INT][TPCH] Implement string expression")
 
-    def endswith(self, suffix: str) -> Expression:
-        raise NotImplementedError("[RUST-INT][TPCH] Implement string expression")
+    def endswith(self, suffix: str | Expression) -> Expression:
+        if not isinstance(suffix, str) and not isinstance(suffix, Expression):
+            raise ValueError(f"Expected a suffix which is either a string or an Expression, received: {suffix}")
+        suffix_expr = lit(suffix) if isinstance(suffix, str) else suffix
```

Comment:
We use `Expression._to_expression(suffix)` in our other functions for handling literals | Expressions